src/txn/txn_recover.c

```diff
@@ -974,8 +1005,19 @@ __wt_txn_recover(WT_SESSION_IMPL *session, const char *cfg[])
          */
         WT_ERR(session->iface.checkpoint(&session->iface, "force=1"));
 
-    /* Initialize the connection's base write generation after rollback to stable. */
-    WT_ERR(__wt_metadata_init_base_write_gen(session));
+    /*
+     * To support rollback to stable operation to fix the possible inconsistent checkpoint data on
+     * disk that is possible to be written when eviction runs in parallel to the checkpoint with a
+     * newer transaction snapshot than the checkpoint thread. To do that the transaction ids that
+     * are saved on disk doesn't get reset until the rollback to stable operation to fix the
+     * inconsistent data.
+     *
+     * Upon successful completion of rollback to stable, need to update the connection base write
+     * generation number with the latest write generation from all the checkpointed files to let the
+     * existing transaction ids to reset to avoid transaction visible failures.
```

Comment:
I suggest reordering this sentence to something like:
Rollback to stable may have updated inconsistent transaction ids. Update the
connection base write generation based on these new ids that may have been reset.

Reply:
Fixed as suggested.